File: leetcode/easy/number-of-strings-that-appear-as-substrings-in-word.py
# https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        ans = 0
        for sub in patterns:
            if sub in word:
                ans += 1
        return ans

File: leetcode/easy/number-of-strings-that-appear-as-substrings-in-word.ts
// https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

// MINE
function numOfStrings(patterns: string[], word: string): number {
  // return patterns.filter(sub => word.includes(sub)).length

  let ans = 0
  for (const sub of patterns) {
    if (word.includes(sub)) ++ans
  }
  return ans
}

// ANSWER
